code/modules/mob/living/basic/space_fauna/carp/carp_ai_rift_actions.dm

```diff
@@ -52,21 +47,19 @@
  */
 /datum/ai_behavior/make_carp_rift
 
+/*
 /datum/ai_behavior/make_carp_rift/setup(datum/ai_controller/controller, ability_key, target_key)
-	var/datum/weakref/weak_action = controller.blackboard[ability_key]
-	var/datum/action/cooldown/mob_cooldown/lesser_carp_rift/ability = weak_action?.resolve()
+	var/datum/action/cooldown/mob_cooldown/lesser_carp_rift/ability = controller.blackboard[ability_key]
 	if (!ability)
 		return FALSE
-	var/datum/weakref/weak_target = controller.blackboard[target_key]
-	var/atom/target = weak_target?.resolve()
-	return target
+
+	return controller.blackboard[target_key]
```

Comment:
were you planning on removing this proc? should probably delete it outright than leaving it commented out.